### PR TITLE
Vendor pandora sdk identity related types

### PR DESCRIPTION
**Pull request by magodo**
This PR defines 5 types of identity related types that exist in azure Swagger spec in the provider, together with the cast to and cast from method with the `ExpandedConfig`. This is to aid the Pandora SDK custom types for identity.

### CODE REVIEW

File: azurerm/internal/identity/schema.go [hunk 1]
```diff
@@ -14,10 +14,10 @@ const (
 type ExpandedConfig struct {
 	// Type is the type of User Assigned Identity, either `None`, `SystemAssigned`, `UserAssigned`
 	// or `SystemAssigned, UserAssigned`
-	Type                    string
-	PrincipalId             *string
-	TenantId                *string
-	UserAssignedIdentityIds *[]string
+	Type                    string   `tfschema:"type"`
+	PrincipalId             string   `tfschema:"principal_id"`
+	TenantId                string   `tfschema:"tenant_id"`
```
Comment (tombuildsstuff) [inline azurerm/internal/identity/schema.go:19]: Whilst this is fine, some of the other service packages (shown below) are expecting `*string` and `*[]string` - so we'll need to update those - I believe this'll be fixed by a rebase?

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hey @magodo

Thanks for this PR - apologies for the delayed review here.

Taking a look through the changes to the identity package looks fine here, but would you mind rebasing this to include the changes from master so that this no longer conflicts?

Thanks!

**Comment by magodo** [comment]
@tombuildsstuff Thank you for your review! I've resolved the comments, please take another look!

**Review by tombuildsstuff — APPROVED** [review]
LGTM - thanks for rebasing this / apologies for the delayed review here @magodo

**Pull request merged**